File: python/my_sqrt.py
def my_sqrt(x):
    guess1 = 0
    guess2 = 1
    x0 = 10
    guess = 0
    
    while int(guess1*1000)/1000 != int(guess2*1000)/1000:
        guess = x0 - (((x0 ** 2) - x)/(2 * x0))
        guess1 = guess2
        guess2 = guess
        x0 = guess

    return int(guess*1000)/1000

File: c/my_sqrt.c
#include <stdio.h>

float mySqrt(int x){

    float guess1 = 0;
    float guess2 = 1;
    float x0 = 10;
    float guess;

    while (guess1 != guess2){
        guess = x0 - ((x0 * x0) - x)/(2 * x0);
        guess1 = guess2;
        guess2 = guess;
        x0 = guess;
    }
    return guess2;

}

int main(){
    int x = 4;
    float result = mySqrt(x);
    printf("%f\n", result);
    return 0;
}